src/textual/_time_to_completion.py

```diff
@@ -0,0 +1,301 @@
+"""Code to help calculate the estimated time to completion of some process."""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+from time import monotonic
+
+from rich.repr import Result
+from typing_extensions import Self
+
+
+@dataclass
```

Comment:
Can this be a NamedTuple? As a tuple it would be a little more memory efficient and hashable by default.

Reply:
Come to think of it, can't think of a reason not to make it so.